molecule/provisioners/libvirtprovisioner.py

```diff
@@ -0,0 +1,923 @@
+
+    def _populate_image(self, instance):
+        """
+        Get this instance's image config.
+
+        Any config specified for this instance is merged with (overriding) any config that already exists for an image of the same name.
+
+        :param: a dict of an instance
+        :return: the final image dict for that instance
+        """
+        image = instance.get(
+            'image', self.molecule.config.config['libvirt']['images'][0])
+        for k in ['source', 'ssh_user', 'ssh_key']:
+            image[k] = image.get(k, filter(
+                lambda imgname: imgname['name'] == image['name'],
+                self.molecule.config.config['libvirt']['images'])[0][k])
+        for index, inst in enumerate(self.molecule.config.config['libvirt'][
+                'instances']):
+            if inst['name'] == instance['name']:
+                if 'image' in self.molecule.config.config['libvirt'][
+                        'instances'][index]:
+                    self.molecule.config.config['libvirt']['instances'][index][
+                        'image'].update(image)
+                else:
+                    self.molecule.config.config['libvirt']['instances'][index][
+                        'image'] = image
+        return image
+
+    def up(self, no_provision=True):
+        """
+        Up or define/up libvirt instances.
+
+        :param: no_provision is not meaningful for libvirt instances
+        """
+        for net in self.molecule.config.config['libvirt']['networks']:
+            self._up_network(net)
+        pool = self._up_pool()
+        vols = pool.listAllVolumes()
+        domains = self._libvirt.listAllDomains()
+        wait = True
+        for instance in self.instances:
+            image = self._populate_image(instance)
+            imagefile = image['source'].split('/')[-1]
+            # Ensure that the image is available
+            if not os.path.exists(
+                    os.path.join(self._pool_path, image['name'] + '.img')):
+                if not os.path.exists(
+                        os.path.join(self._sources_path, imagefile)):
+                    self._fetch(image['source'], imagefile)
+                if imagefile.endswith('.box'):
+                    self._unpack_box(image, imagefile)
+
+            # Is there an existing libvirt volume for this instance?
+            try:
+                vol_found = pool.storageVolLookupByName(instance['name'])
+                if vol_found and not vol_found.isActive():
+                    vol_found.create()
+                return vol_found  # existing volume is now running
+            except libvirt.libvirtError as e:
+                self._create_volume(pool, instance)
+            # TODO: Are all this instance's interface's networks available?
+            # Is there an existing libvirt domain defined for this instance?
+            dom_found = False
+            for dom in domains:
+                if not dom_found:
```

Comment:
Since you're only looking for *one* dom, you can greatly simplify the flow in this function by using `next` (and thus removing the loop, the break, all of that):
`dom = next((domain in domains if domain.name() == instance['name']), None)`
[Note the usage of a default value: `None`]